sql/api/src/main/scala/org/apache/spark/sql/catalyst/trees/origin.scala

```diff
@@ -76,6 +76,12 @@ object CurrentOrigin {
       value.get.copy(line = Some(line), startPosition = Some(start)))
   }
 
+  def setPySparkErrorContext(pysparkFragment: String, pysparkCallSite: String): Unit = {
+    val tupleInfo = (pysparkFragment, pysparkCallSite)
+    value.set(
+      value.get.copy(pysparkErrorContext = Some(tupleInfo)))
+  }
+
```

Comment:
cc @HyukjinKwon @cloud-fan 

Try reusing the existing `CurrentOrigin` here so we don't need to add separate ThreadLocal and also don't need to add helper functions for each existing methods.

Some test cases are still need to be addressed, but I believe the overall structure is ready to review.